src/powergim/investment_model.py

```diff
@@ -199,6 +202,39 @@ def bounds_load_shed(model, consumer, period, time):
             bounds=bounds_load_shed,
         )
 
+        # Flexible load (shiftable and price sensitive)
+        def bounds_load_flex_shift(model, consumer, period, t):
+            ub = 0
+            lb = 0
+            if self.has_load_flex_shift:
+                flex_frac = self.parameters["load_flex_shift_frac"][period]
+                flex_onoff = self.parameters["load_flex_shift_max"][period]
+                demand_avg = 0
+                previous_periods = [p for p in self.s_period if p <= period]
+                for p in previous_periods:
+                    demand_avg += self.grid_data.consumer.at[consumer, f"demand_{p}"]
+                lb = -demand_avg * flex_frac
+                ub = demand_avg * flex_frac * flex_onoff
```

Comment:
The idea is that  in any given time, the load is minimum, (1-flex_frac), but the maximum can be more than (1+frec_frac) because e.g. shifted load can be piled on top of each other, to some extent (given by the flex_onoff parameter). For example load may be shifted from time1 and time2 and both added to time3.